core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1211,16 +1285,10 @@ write_files:
                         break
                       fi
 
-                      if [ $((secs % SG_POLL_INTERVAL)) -eq 0 ]; then
-                        # Reset counter to avoid overflow
-                        secs=0
-
-                        # AutoScalingGroup lifecycle check
-                        state=$(asg describe-auto-scaling-instances --region "${REGION}" --instance-ids "${INSTANCE_ID}" | jq -r '.AutoScalingInstances[].LifecycleState')
-                        if [ "${state}" = Terminating:Wait ]; then
-                          termination_source=asg
-                          break
-                        fi
+                      # Termination annotation check
+                      termination_source=$(kubectl get nodes "${NODE_NAME}" -o json | jq -r '.metadata.annotations["kube-aws.coreos.com/node-termination-source"]')
```

Comment:
ConfigMap can be written to a file by kubelet and be updated anytime it changes in APIServer via https://kubernetes.io/docs/concepts/storage/volumes/#example-pod-with-a-secret-a-downward-api-and-a-configmap

wait for signal is then just stat and jq check, something like `[ "$(stat -c %y configmap)" != $PREV_MTIME ] && jq -e '(len(select(.path.to.node == $mynode))) > 0'`